### PR TITLE
bigdft : add v1.9.5

### PR DESCRIPTION
**Pull request by viperML**
- Bump all the packages from BigDFT to 1.9.5.
- Add `bigdft-liborbs` as a dependency - it wasn't needed before?
- Removes `ntpoly` variant from `bigdft-chess`. I believe this is a hard-dependency now.
- Compiles `libgain` with `-fPIC`

### CODE REVIEW

**Comment by viperML** [comment]
Pinging people who touched bigdft in the commit history: @danielecesarini @iamashwin99

**Review by tldahlgren — DISMISSED** [review]
Confirmed new version sha256s, homepage+branch+sha256 of new package, and version of `bigdft-suite`.

**Comment by tldahlgren** [comment]
@iamashwin99 As a recent significant contributor to nearly every package here, are you interested in reviewing this PR?

**Comment by viperML** [comment]
- Made `bigdft-liborbs` a dependency of only `bigdftcore >= 1.9.5` (I don't want to touch the older versions)

File: var/spack/repos/builtin/packages/bigdft-atlab/package.py
Comment (iamashwin99) [inline var/spack/repos/builtin/packages/bigdft-atlab/package.py:17]: I have verified that the hash for all the packages is correct.

File: var/spack/repos/builtin/packages/bigdft-atlab/package.py
Comment (iamashwin99) [inline var/spack/repos/builtin/packages/bigdft-atlab/package.py:44]: Could we reduce the complexity by maybe specifying it this way:
```suggestion
    with when("@1.9.0:"):
        depends_on(f"bigdft-futile@{self.version}")

```
I haven't tested this out but it seems plausible.

File: var/spack/repos/builtin/packages/bigdft-chess/package.py
Comment (iamashwin99) [inline var/spack/repos/builtin/packages/bigdft-chess/package.py:18]: Just curious if you tried compiling `1.9.4` and `1.9.3`  without any other changes, if so we could consider adding their hashes as well.
https://gitlab.com/l_sim/bigdft-suite/-/releases

**Review by iamashwin99 — COMMENTED** [review]
Thanks @viperML for the new package and version bumps,
While I haven't yet compiled all the packages for the new version my self, here are some of my thoughts.

**Comment by viperML** [comment]
Thank you for the reviews, I'll look at them as soon as possible -- it's also my first contribution to Spack.

File: var/spack/repos/builtin/packages/bigdft-atlab/package.py
Comment (viperML) [reply var/spack/repos/builtin/packages/bigdft-atlab/package.py:44]: Should I create an `__init__` for the class, such that we have `self` in scope?

**Comment by viperML** [comment]
`bigdft-suite` from 1.9.0 to 1.9.2 fail to build on my machine for the unmodified `develop` branch (caused by different components).

It could be of interest either to fix the packages in a follow-up PR, or remove the versions.

File: var/spack/repos/builtin/packages/bigdft-atlab/package.py
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/bigdft-atlab/package.py:44]: @viperML:

Maybe adding `__init__` works, but I haven't seen that in a recipe - check maybe if it is done in other recipes.

You can use a range of minor versions like this:
```py
    for v in range(30, 38):
        depends_on(f"py-pennylane-lightning@0.{v}:", type=("build", "run"), when=f"@0.{v}:")
```
@viperML Please ping me or other `spack` maintainers by mentioning us to get your change reviewed again.

@viperML: Also, I recommend submit a tiny PR that only updates a base dependency (e.g. like `libgain` or `ntpolt` - as long as these update don't depend on each other) in new PRs to get on top of the review queue and make approval and fixes easier than with a big PR like this.

I'm resolving this discussion because I prefer to merge it as is.

**Review by bernhardkaindl — COMMENTED** [review]
> bigdft-suite from 1.9.0 to 1.9.2 fail to build on my machine for the unmodified develop branch (caused by different components).
>
> It could be of interest either to fix the packages in a follow-up PR, or remove the versions.

Just leave these alone (for now) and don't bother with changes that you don't need at the moment. Instead of focus on submitting new PRs that update those base dependencies that are not dependent on other updates (see my other review comment).

These new small singe-package PRs would be on the review queue and have a higher chance of being fast to approve and are also easier to review and fix than a big submission.

File: var/spack/repos/builtin/packages/bigdft-chess/package.py
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/bigdft-chess/package.py:18]: @iamashwin99 I'd like to resolve this conversation to allow merging this PR.

Issues/questions with the versions will resolve itself as they become even more obsolete.

Comment (iamashwin99) [reply var/spack/repos/builtin/packages/bigdft-chess/package.py:18]: Sure! This was just a thought; I wanted to see if the other versions also compile with the new changes.
We can also ignore the older versions for now.

**Review by bernhardkaindl — DISMISSED** [review]
Approving as submitted by the maintainer @viperML and @iamashwin99 as recent active contributor. I did not see the review comments anywhere near blocking the merge.

File: var/spack/repos/builtin/packages/py-bigdft/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-bigdft/package.py]: Versions 1.9.4-1.9.5 require a newer version of `python`:

```
requires-python = ">=3.6"
```

There appear to be missing dependencies as well per their `setup.py` files.  For 1.9.3:

```
      install_requires=['ase', 'matplotlib', 'numpy'],
```

and versions 1.9.4-1.9.5:

```
      install_requires=['numpy', 'scipy', 'PyFutile'],'
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the new version sha256s (when provided), homepage of new package, branch of new package, and presence of new versions of `bigdft-suite`.  However, there appear to be dependency deficiencies for the new versions of `py-bigdft` (see comments).

**Review by bernhardkaindl — APPROVED** [review]
@tldahlgren: I saw that the changes you requested are pushed now:

https://github.com/spack/spack/pull/45270/commits/5cea9a77f4be7017c3b9ae5d356a08d45dd4691c#diff-7aaebb5a0a572d857c94b753f81b95d74c9b1ece13596f6057d14cd321b22a23R30

**Review by tldahlgren — APPROVED** [review]

**Comment by viperML** [comment]
Thank you for the support

**Pull request merged**
